photosynthesis_metrics/fid.py

```diff
@@ -52,19 +46,62 @@ def __compute_fid(mu1: float, sigma1: np.ndarray, mu2: float, sigma2: np.ndarray
         covmean = covmean.real
 
     tr_covmean = np.trace(covmean)
-    return (diff.dot(diff) + np.trace(sigma1) +
-            np.trace(sigma2) - 2 * tr_covmean)
-
-
-def compute_fid(gt_stack: np.ndarray, denoised_stack: np.ndarray) -> float:
-    m_gt, s_gt = compute_statistics(gt_stack)
-    m_denoised, s_denoised = compute_statistics(denoised_stack)
-
-    fid_gt_denoised = __compute_fid(m_gt, s_gt, m_denoised, s_denoised)
-    return fid_gt_denoised
+    return (diff.dot(diff) + np.trace(sigma1) + np.trace(sigma2) - 2 * tr_covmean)
+
+def _compute_statistics(samples: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    r"""Calculates the statistics used by FID
+    Args:
+        samples:  Low-dimension representation of image set.
+            Shape (N_samples, dims) and dtype: np.float32 in range 0 - 1
+    Returns:
+        mu: mean over all activations from the encoder.
+        sigma: covariance matrix over all activations from the encoder.
+    """
+    mu = np.mean(samples, axis=0)
+    sigma = np.cov(samples, rowvar=False)
+    return mu, sigma
 
 
-def compute_statistics(stack: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
-    mu = np.mean(stack, axis=0)
-    sigma = np.cov(stack, rowvar=False)
-    return mu, sigma
+def compute_fid(x: torch.Tensor, y: torch.Tensor) -> float:
+    r"""Numpy implementation of the Frechet Distance.
+    Fits multivariate Gaussians: X ~ N(mu_1, sigm_1) and Y ~ N(mu_2, sigm_2) to image stacks.
+    Then computes FID as d^2 = ||mu_1 - mu_2||^2 + Tr(sigm_1 + sigm_2 - 2*sqrt(sigm_1*sigm_2)).
+
+    Args:
+        x: Samples from data distribution. Shape (N_samples, data_dim), dtype: torch.float32 in range 0 - 1.
+        y: Samples from data distribution. Shape (N_samples, data_dim), dtype: torch.float32 in range 0 - 1
+
+    Returns:
+    --   : The Frechet Distance.
+    """
+    m_pred, s_pred = _compute_statistics(x.numpy())
+    m_targ, s_targ = _compute_statistics(y.numpy())
+
+    score = __compute_fid(m_pred, s_pred, m_targ, s_targ)
+    return score
+
+
+class FID(BaseFeatureMetric):
+    r"""Creates a criterion that measures Frechet Inception Distance score for two datasets of images
+    See https://arxiv.org/abs/1706.08500 for reference.
+    """
+    def __init__(self):
+        super(FID, self).__init__()
+        self.compute = compute_fid
+
+    def forward(self, predicted_features: torch.Tensor, target_features: torch.Tensor) -> torch.Tensor:
```

Comment:
Wrong type annotation. This forward is annotated such that it returns `torch.Tensor`, but `compute_fid` from which you obtain the actual score returns `float`. That is why this method always returns `float`.

```suggestion
    def forward(self, predicted_features: torch.Tensor, target_features: torch.Tensor) -> float:
```